src/internal/components/ThemeEditor.tsx

```diff
@@ -42,8 +42,9 @@ export const ThemeEditor = (props: ThemeEditorProps) => {
   >();
   const [puckInitialHistoryFetched, setPuckInitialHistoryFetched] =
     useState<boolean>(false);
-  const [themeInitialHistory, setThemeInitialHistory] =
-    useState<ThemeSaveState>({ history: [{}], index: 0 });
+  const [themeInitialHistory, setThemeInitialHistory] = useState<
```

Comment:
Not you, but I don't think the term "initial" history makes sense for themes. For Puck it does because you need to initialize Puck with the initial state of the history and then Puck deals with the history stuff internally, so we need multiple variables for it. For theme, we're maintaining that state ourselves so we basically just need to load the themeHistory and then save it.